prefect_kubernetes/credentials.py

```diff
@@ -1,13 +1,22 @@
 """Module for defining Kubernetes credential handling and client generation."""
 
 from contextlib import contextmanager
-from typing import Generator, Optional
+from typing import Generator, Optional, Union
 
 from kubernetes import config as kube_config
-from kubernetes.client import ApiClient, AppsV1Api, BatchV1Api, CoreV1Api
+from kubernetes.client import ApiClient, AppsV1Api, BatchV1Api, Configuration, CoreV1Api
 from kubernetes.config.config_exception import ConfigException
 from prefect.blocks.core import Block
 from prefect.blocks.kubernetes import KubernetesClusterConfig
+from prefect.utilities.collections import listrepr
+
+KubernetesClient = Union[AppsV1Api, BatchV1Api, CoreV1Api]
+
+K8S_CLIENT_TYPES = {
+    "apps": AppsV1Api,
+    "batch": BatchV1Api,
+    "core": CoreV1Api,
+}
```

Comment:
Just a thought; what if we make this an enum?

Reply:
that would also make sense, after talking to alex though I think I'll just keep these as literals

Reply:
Literals and enums fill a very similar role, but literals are simpler overall. If the additional features that enums offer aren't necessary, I think we should use literals, and this looks like a reasonable place to use literals.